docker-compose.yml

```
@@ -0,0 +1,60 @@
+version: "3.8"
+
+services:
+  charterafrica:
+    depends_on:
+      mongodb:
+        condition: service_started
+    build:
+      context: .
+      dockerfile: Dockerfile.charterafrica
+      args:
+        MONGO_URL: mongodb://root:rootpassword@host.docker.internal:27017/charterafrica?authSource=admin&directConnection=true
+        PAYLOAD_SECRET_KEY: ${PAYLOAD_SECRET_KEY}
+        SENTRY_AUTH_TOKEN: ${SENTRY_AUTH_TOKEN}
+        SENTRY_ORG: ${SENTRY_ORG}
+        NEXT_PUBLIC_SENTRY_DSN: ${CHARTERAFRICA_SENTRY_DSN}
+        SENTRY_ENV: ${SENTRY_ENV}
+        SENTRY_PROJECT: ${SENTRY_PROJECT}
+    ports:
+      - 3000:3000
+    container_name: charterafrica
+    env_file:
+      - apps/charterafrica/.env.local
+    networks:
+      - net
+
+  mongodb:
+    container_name: mongodb
+    image: mongo:6.0.13
+    environment:
+      MONGO_INITDB_ROOT_USERNAME: root
+      MONGO_INITDB_ROOT_PASSWORD: rootpassword
+    ports:
+      - 27017:27017
+    volumes:
+      - ./contrib/docker/db:/data/db
+      - ./mongo-keyfile:/etc/mongo-keyfile
+      - ./contrib/docker/scripts/dbInit.js:/docker-entrypoint-initdb.d/dbInit.js:ro
+    restart: always
+    healthcheck:
+      test: echo 'db.runCommand("ping").ok' | mongosh mongodb:27017/charterafrica --quiet
+      interval: 10s
+      timeout: 5s
+      retries: 3
+    command:
+      [
+        "mongod",
+        "--replSet",
+        "dbrs",
+        "--bind_ip_all",
+        "--keyFile",
+        "/etc/mongo-keyfile",
+      ]
```

```suggestion
    command: ["sh", "-c", "chmod 600 /etc/mongo-keyfile && exec mongod --replSet dbrs --bind_ip_all --keyFile /etc/mongo-keyfile"]
```
Mongo wont start if the keyfile has the wrong permission. From [here](https://stackoverflow.com/a/14791571)

---

@koechkevin I see that you have updated it in the description, but I think the `chmod` command should be run inside the container.

---

Updated the description to capture this @kelvinkipruto